### PR TITLE
test: add integ test for permissions boundary

### PR DESCRIPTION
**Pull request by huanjani**
This integ test ensures that as we add CFN templates and/or IAM roles to CFN templates, we also conditionally attach a permission boundary if present in the app.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/permissions_boundary_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+func TestPermissions_Boundary(t *testing.T) {
+	t.Run("every CloudFormation template must contain conditional permissions boundary field for all IAM roles", func(t *testing.T) {
+		cmd := exec.Command("find", "templates", "-name", "*yml")
+		output, err := cmd.Output()
```
Comment (efekarakus) [inline internal/pkg/template/permissions_boundary_integration_test.go]: Is there a reason why we preffered using `find`? would this work on different platforms like Windows 🤔 
Can we leverage `os.ReadDir` and walk through the files that way?

File: internal/pkg/template/permissions_boundary_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		for _, file := range files {
+			contents, err := os.ReadFile(file)
+			require.NoError(t, err, "should read file")
+			IAMRoles := bytes.Count(contents, []byte("AWS::IAM::Role"))
+			permissionsBoundaryFields := bytes.Count(contents, []byte("PermissionsBoundary:"))
+				
+			totalIAMRoles += IAMRoles
+			totalPermissionsBoundaryFields += permissionsBoundaryFields
+		}
+		require.True(t, totalIAMRoles == totalPermissionsBoundaryFields, "number of IAM roles does not equal number of permissions boundary fields")
```
Comment (efekarakus) [inline internal/pkg/template/permissions_boundary_integration_test.go]: I think with the current implementation we won't be able to tell which file doesn't have the `PermissionsBoundary:` set.

Can we fail early instead in the for-loop writing exactly which file (the filepath) has the permissions boundary missing? 
For example, try removing one in the templates and see the error message you get from the test to see if that would help the next engineer 😂 

We could even potentially figure out which IAM role doesn't have PermissionsBoundary set by analyzing the line numbers but it's prob over engineering

File: internal/pkg/template/permissions_boundary_integration_test.go (hunk 1 above)
Comment (huanjani) [reply internal/pkg/template/permissions_boundary_integration_test.go]: I switched it to `filepath.WalkDir`...what do you think?

File: internal/pkg/template/permissions_boundary_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestPermissions_Boundary(t *testing.T) {
+	t.Run("every CloudFormation template must contain conditional permissions boundary field for all IAM roles", func(t *testing.T) {
+		err := filepath.WalkDir("templates", func(path string, di fs.DirEntry, err error) error {
+			contents, _ := os.ReadFile(path)
```
Comment (efekarakus) [inline internal/pkg/template/permissions_boundary_integration_test.go]: nit:
```suggestion
			contents, err := os.ReadFile(path)
			require.NoError(err, "read file at %s", path)
```

File: internal/pkg/template/permissions_boundary_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+			IAMRoles := bytes.Count(contents, []byte("AWS::IAM::Role"))
+			permissionsBoundaryFields := bytes.Count(contents, []byte("PermissionsBoundary:"))
+
+			msg := fmt.Sprintf("number of IAM roles (%d) does not equal number of permissions boundary fields (%d) in file '%s'", IAMRoles, permissionsBoundaryFields, path)
+			require.True(t, IAMRoles == permissionsBoundaryFields, msg)
```
Comment (efekarakus) [inline internal/pkg/template/permissions_boundary_integration_test.go]: nit:
```suggestion
			require.Equal(t, roleCount, fieldCount, "number of IAM roles does not equal number of permissions boundary fields in file %s", path)
```

**Review by efekarakus — APPROVED** [review]
Looks great! just some minor suggestions to leverage the `testify` lib feel free to remove the label after

File: internal/pkg/template/permissions_boundary_integration_test.go (hunk 3 above)
Comment (huanjani) [reply internal/pkg/template/permissions_boundary_integration_test.go]: Ah, I had taken that `err` out because `ReadFile` was returning errors when trying to read directories, like `read templates: is a directory`

Comment (huanjani) [reply internal/pkg/template/permissions_boundary_integration_test.go]: added `IsDir()`

**Review by bvtujo — APPROVED** [review]
`WalkDir` is a nifty function!
